### DISCUSSION TITLE
Lag occurs when dragging the element with links

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2570] [Q&A]
### Introduction

I have created around 290 rectangles and created one  circle  and connected each rectangle with the circle
When I try to drag the circle, lag occurs and the entire screen freezes. 
is there any way to fix this issue?

![video](https://github.com/clientIO/joint/assets/117301232/e94790a2-74a8-437f-84f1-5acbe14f9e64)

https://jsfiddle.net/reshin_pulapaddy/scuexpty/6/

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
1. Upgrade to version 4. It will force you to get rid of the legacy `dia.Link` slow links. This is the most **important** point.
2. You can specify the maximum number of updates in a single animation frame: `paper.unfreeze({ batchSize: 100 })`
3. If you expect elements in your app to be connected by hundred of links, you might want to implement dragging previews as shown in [DataMapping](https://www.jointjs.com/demos/data-mapping) application.
